### ISSUE TITLE
How to use GPU head instead of SPICE/VNC?

### ISSUE THREAD

**Issue by renne** [01org/gvt-linux#89]
Hi,

is it possible to assign a GPU head to a KVM guest instead of using a SPICE/VNC?

**Comment by snir911** [comment]
Hi,

Yes, on local display, but you can still use spice/vnc (configured a bit differently) to connect to it 
see https://github.com/intel/gvt-linux/wiki/Dma_Buf_User_Guide
(seems not really up to date)

**Comment by renne** [comment]
Is it possible to write the graphic output directly into the framebuffer of a head without Xserver and network on the hypervisor?

Examples:

Core i5-7500 + B150

- DVI-D = Seat0
- DP++ 1 = Seat1
- DP++ 2 = Seat2

Celeron G4900 + B360

- DVI-D = Host
- DP++ 1 = KVM guest 1
- DP++ 2 = KVM guest 2

**Comment by snir911** [comment]
Do you mean that guest has no Xserver? If i understood you correctly you can do that with spice

**Comment by renne** [comment]
In the Core i5-7500 + B150 scenario the host runs a Xserver per seat. So three physical GPUs have to be emulated for a 3x multi-seat system.

In the Celeron G4900 + B360 scenario the host is console only and a Xserver runs inside the KVM guest (e.g. [LibreElec](https://libreelec.tv/)).

**Comment by cagnulein** [comment]
@renne yes it's possible You can run a X session for each output and running on X session a spice session of X VM

so for example:

Core i5-7500 + B150

DVI-D = Seat0 --> X :0
DP++ 1 = Seat1 --> X :1
DP++ 2 = Seat2 --> X :2
Celeron G4900 + B360

DVI-D = Host  --> X :0
DP++ 1 = KVM guest 1 --> X :1
DP++ 2 = KVM guest 2 --> X :2

Is it ok for you?

**Comment by renne** [comment]
I'd like to avoid network overhead and network dependency. A virtual GPU with hardware acceleration (especially h.264, h.265, VP9) per head would be great. See above [comment](https://github.com/intel/gvt-linux/issues/89#issuecomment-483225721).

**Comment by cagnulein** [comment]
@renne https://looking-glass.hostfission.com/ ?
